Sources/Views/LoadingView.swift

```diff
@@ -0,0 +1,100 @@
+//
+//  LoadingView.swift
+//  PovioKit
+//
+//  Created by Toni Kocjan on 04/11/2021.
+//  Copyright © 2021 Povio Labs. All rights reserved.
+//
+
+import UIKit
+
+/// Adapt any view into a loading view.
+open class LoadingView<V: UIView>: UIControl {
+  public let view: V
+  public let activityIndicator: UIActivityIndicatorView
+  let animations: Animations?
```

Comment:
This seems like overkill, nesting views. Could we just make a protocol and a method that will start/stop the animation? At start, we could insert the indicator, on the stop we should remove it 🤷‍♂️ 🤔

Reply:
I would argue that adding (together with layout-ing) and removing a subview is much more expensive than just hiding a view.
Not sure how you would solve this with a protocol. Can you give an example?

Reply:
What you are currently doing here is wrapping a "view" with another view, creating an `activityIndicator` inside and maybe all those things will not be needed at all in the lifecycle of the view.

What I would do is to just add `activityIndicator` to the view when needed, via extension method, smth like that:
```swift
extension UIView {
  func showLoader() {
    // add subview
    // set constraints
    // set tag
    // start loading
  }

  func hideLoader() {
    // find subview by tag
    // stop loading
    // remove
  }
}
```
Sure, this approach is a bit unreliable since we are relying on the `tag` which can be easily overridden. But I just wanted to present the idea, how this "loader" thing could be lightweight and only created when needed.
What do you think?

Reply:
Doing it in such a way, wouldn't the activity indicator cover the content?

![image](https://user-images.githubusercontent.com/15092411/141486344-b3f589b2-3af8-4e9e-9bce-6f51adbd116d.png)

Reply:
Isn't that how it suppose to be?

Reply:
At least the label should be hidden, I think. Otherwise, it looks kind of silly. 🤷 

https://user-images.githubusercontent.com/15092411/141788708-d6bd6b08-fe38-4978-a5ae-a532fcda1fe0.mov

With this implementation you can easily add an activity indicator to an image view with a nice fade animation:

https://user-images.githubusercontent.com/15092411/141788775-93fbe5ac-a19a-419e-b8be-c12cebb34fe6.mov

_p.s. I don't have a strong opinion on this ticket. We can go with the suggested lightweight implementation instead._

Reply:
Good point 👌 
Neither do I. Let's wait for other guys opinions.

Reply:
I tried to use this in a simple case, where I wanted to show an activity indicator over the whole screen e.g. `self.view`, It did give me some unexpected results (screen went black), and also as @borut-t mentioned the nesting when used incorrectly (such as in my case with multiple ui elements) could produce a negative performance impact.